features/cellular/framework/API/CellularContext.h

```diff
@@ -211,6 +212,12 @@ class CellularContext : public CellularBase {
      */
     virtual void set_file_handle(FileHandle *fh) = 0;
 
+    /** Get the ATHandler instance currently in use. Can be used to send ad-hoc AT commands to
+     *  the modem for debugging etc. purposes
+     *  @return             Reference to the ATHandler in use
+     */
+    virtual ATHandler &get_at_handler() = 0;
```

Comment:
CellularContext is not always derived from AT_CellularContext. Any way to use here more generic interface than ATHandler?

Reply:
It's not now clear  that handler reference may die when context is deleted. Should probably use reference counting similar to other interfaces?

Reply:
It might make sense for this thing to return a pointer, so it can return NULL if the implementation doesn't have an `ATHandler` available. It should only be a reference if it is a requirement that an ATHandler is available in every CellularContext.

(For comparison, `NetworkInterface::emacInterface` returns a pointer, as a `NetworkInterface` obviously doesn't have to be an `EMACInterface`. But  `EMACInterface::get_emac()` returns a reference, as an `EMACInterface` must have an `EMAC`.)

If you're returning an `ATHandler` you were given and you're using for your context, then presumably `CellularContext` has no lifetime view on it. Reference counting would be down to `ATHandler` itself - if it has a reference count, the `CellularContext` should increment it before passing it on. Or you could use `mbed::SharedPtr`. Not sure how well that works - never used it myself. Or just wait for C++11 `std::shared_ptr`. Coming real soon, maybe.